python/mlx/nn/losses.py

```diff
@@ -62,7 +101,9 @@ def cross_entropy(
 
     # Apply weights if provided
     if weights is not None:
-        if weights.shape != targets.shape:
+        if (targets_as_probs and weights.shape != _drop_dim(targets.shape, axis)) or (
+            not targets_as_probs and weights.shape != targets.shape
+        ):
             raise ValueError(
                 f"Weights with shape {weights.shape} is not the same as "
                 f"targets with shape {targets.shape}."
```

Comment:
Please update this error message since now the weight should have the same shape as the loss but not necessarily the target.